### PR TITLE
Added 'Blue Moon Inn'

### PR DESCRIPTION
**Pull request by nicknick-io**
Let's get our first pub in the game.

### CODE REVIEW

File: src/plugins/npcs/varrock/blue-moon-inn-plugin.ts [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+                            player => [Emote.VERY_SAD, `Oh dear. I don't seem to have enough money.`],
+                        ]);
+                    }
+
+                    if (hasCoins && (player.inventory.amountInStack(index) >= 2)) {
+
+                        const amount = player.inventory.amountInStack(index);
+                        // Check inventory.
+                        if (!player.inventory.hasSpace()) {
+                            player.sendMessage(`You don't have enough space in your inventory.`);
```
Comment (Promises) [inline src/plugins/npcs/varrock/blue-moon-inn-plugin.ts:33]: Could we move this to the player object as a public function?

File: src/plugins/npcs/varrock/blue-moon-inn-plugin.ts [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+                        // Take the coins
+                        player.inventory.remove(index);
+                        if ((amount - 2) !== 0) {
+                            player.inventory.add({
+                                itemId: itemIds.coins,
+                                amount: (amount - 2)
+                            });
+                        }
+
+                        // Give the cabbage.
```
Comment (Promises) [inline src/plugins/npcs/varrock/blue-moon-inn-plugin.ts]: Cabbage?

File: src/plugins/npcs/varrock/blue-moon-inn-plugin.ts (hunk 1 above)
Comment (nicknick-io) [reply src/plugins/npcs/varrock/blue-moon-inn-plugin.ts:33]: That sounds like work - I just want beer :(

File: src/plugins/npcs/varrock/blue-moon-inn-plugin.ts (hunk 2 above)
Comment (nicknick-io) [reply src/plugins/npcs/varrock/blue-moon-inn-plugin.ts]: Beer is cabbage when you get one too many :D

**Review by TheBlackParade — APPROVED** [review]

**Pull request merged**